### PR TITLE
PB-776: k8s and cloudfront improvements

### PR DESCRIPTION
**Pull request by ltflb-bgdi**

### CODE REVIEW

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -475,13 +479,30 @@ processors:
...
+
+          // https://gist.github.com/vjt/06b28fbd988788c2a7a71c63dd9163be
+          def header_list = header_string.splitOnToken((String)(char)0x0a);
+
+          for (item in header_list) {
+            def kv = item.splitOnToken(':', 2);
+            if (kv.length == 2) {
+              String key = kv[0].toLowerCase().replace('-','_');
+              header_map.put(key, kv[1]);
+            }
```
Comment (boecklic) [inline packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:499]: in your comment below you say that empty headers are added as well, but from the code it looks like they are not? Or is a header like `Header:` resulting in `kv.length = 2`?

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -676,4 +699,7 @@ on_failure:
       value: pipeline_error
   - append:
       field: error.message
-      value: '{{{ _ingest.on_failure_message }}}'
+      value: >-
+        Processor '{{ _ingest.on_failure_processor_type }}'
+        {{#_ingest.on_failure_processor_tag}}with tag '{{ _ingest.on_failure_processor_tag }}'
+        {{/_ingest.on_failure_processor_tag}}failed with message '{{ _ingest.on_failure_message }}'
```
Comment (boecklic) [inline packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:705]: not sure what `{{#_ingest.on_failure_processor_tag}}` and `{{/_ingest.on_failure_processor_tag}}` does but I assume it's ok.

File: packages/ppbgdi/data_stream/kubernetes_event_logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -148,6 +152,10 @@ processors:
       field: _tmp.regarding.resourceVersion
       target_field: kubernetes.event.regarding.version
       ignore_missing: true
+  - rename:
+      field: _tmp.regarding.reportingInstance
```
Comment (boecklic) [inline packages/ppbgdi/data_stream/kubernetes_event_logs/elasticsearch/ingest_pipeline/default.yml:156]: what is instance referring to here? node or kubernetes object? or sth else?

**Review by boecklic — APPROVED** [review]
:+1: very nice and useful, thanks!

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -475,13 +479,30 @@ processors:
...
+          header_string = header_string.trim();
+          Map header_map = new HashMap();
+
+          // https://gist.github.com/vjt/06b28fbd988788c2a7a71c63dd9163be
+          def header_list = header_string.splitOnToken((String)(char)0x0a);
+
+          for (item in header_list) {
+            def kv = item.splitOnToken(':', 2);
+            if (kv.length == 2) {
+              String key = kv[0].toLowerCase().replace('-','_');
```
Comment (ltshb) [inline packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:497]: @boecklic why changing he header from hyphen to underscore ? Might be a good idea to comment the reason here. Perso I would have kept hyphen for headers but you surely have a good reason for this ?

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (ltflb-bgdi) [reply packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:499]: Aha, what I wanted to say is that if a headers field is sent by cloudfront, the elastic event will get the http.request.headers map, even if the headers field from cloudfront does not contain headers or the headers can not be parsed. From what I undertand, headers are always key/value, thus if not two elements after parsing, they are not added.

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (ltflb-bgdi) [reply packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:705]: This is a new pattern introduced by elastic that will be rolled out to all error handlers. This helps to identify which one of the processors of the same type failed. Adding processor tags are only used in the context of the ingest pipeline itself, e.g. error handlers and metrics and are optional.

File: packages/ppbgdi/data_stream/kubernetes_event_logs/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (ltflb-bgdi) [reply packages/ppbgdi/data_stream/kubernetes_event_logs/elasticsearch/ingest_pipeline/default.yml:156]: Seems to be the node. You may check in the logs. Problem is the logs are appear as they are, without much documentation to be found about it.

File: packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (ltflb-bgdi) [reply packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:497]: The idea was to use underlines, like recommended on other field names. But technically it is not necessary. I can easily revert that.

**Pull request merged**